src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/ExtendedQueryTagVersion.cs

```diff
@@ -0,0 +1,117 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
+{
+    /// <summary>
+    /// ExtendedQueryTag version.
+    /// </summary>
+    /// <remarks>
+    /// We use SQL ROWVERSION to track changes on extended query tags. Every time the tag is changed, a new version is assigned to.
+    /// We use max version amoung all extended query tags as version of ExtendedQueryTagTable, since ROWVERSION keep increasing and never falls back.
+    /// The version is used to indicate if extended query tags we are using are same as ExtendedQueryTagTable.
+    /// </remarks>
+    public struct ExtendedQueryTagVersion : IEquatable<ExtendedQueryTagVersion>, IComparable<ExtendedQueryTagVersion>
+    {
+        /// <summary>
+        /// Get extended query tag version for a list of tag versions.
+        /// </summary>
+        /// <param name="tagVersions">The tag version collection.</param>
+        /// <returns>The tag version.</returns>
+        public static ExtendedQueryTagVersion? GetExtendedQueryTagVersion(IReadOnlyCollection<ExtendedQueryTagVersion> tagVersions)
+        {
+            EnsureArg.IsNotNull(tagVersions, nameof(tagVersions));
+            return tagVersions.Count == 0 ? null : tagVersions.Max();
+        }
+
+        /// <summary>
+        /// Get extended query tag version for a list of query tags.
+        /// </summary>
+        /// <param name="queryTags">The query tags.</param>
+        /// <returns>The tag version.</returns>>
+        public static ExtendedQueryTagVersion? GetExtendedQueryTagVersion(IReadOnlyCollection<QueryTag> queryTags)
+        {
+            EnsureArg.IsNotNull(queryTags, nameof(queryTags));
+            return GetExtendedQueryTagVersion(queryTags
+                .Where(x => x.IsExtendedQueryTag && x.ExtendedQueryTagStoreEntry.Version.HasValue)
+                .Select(x => x.ExtendedQueryTagStoreEntry.Version.Value)
+                .ToList());
+        }
+
+        private readonly ulong _version;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExtendedQueryTagVersion"/> class.
+        /// </summary>
+        /// <param name="version">The version.</param>        
+        public ExtendedQueryTagVersion(byte[] version)
```

Comment:
Do we need this type or can we simply use `ulong` directly?

Reply:
Our code generator returns `byte[]` for `ROWVERSION`,  I need some code to convert that to `ulong`, feel the constructor might be a good place.